src/Components/CreateImageWizard/formComponents/Packages.js

```diff
@@ -156,60 +163,99 @@ const Packages = ({ defaultArch, ...props }) => {
+        const newPackagesChosen = [];
 
-        const updatedSourcePackages = sourcePackages.filter((pack) => {
+        const updatedPackagesAvailable = packagesAvailable.filter((pack) => {
             if (pack.selected) {
                 pack.selected = false;
-                destinationPackages.push(pack);
+                pack.isHidden = isHidden(filterChosen.current, pack);
+                newPackagesChosen.push(pack);
                 return false;
             }
 
             return true;
         });
 
-        if (fromAvailable) {
-            sortPackages(updatedSourcePackages);
-            setPackagesChosen(destinationPackages);
-            // set the steps field to the current chosen packages list
-            change(input.name, removePackagesDisplayFields(destinationPackages));
-        } else {
-            setPackagesChosen(updatedSourcePackages);
-            sortPackages(packagesAvailable);
-            // set the steps field to the current chosen packages list
-            change(input.name, removePackagesDisplayFields(updatedSourcePackages));
-        }
+        const updatedPackagesChosen = [ ...newPackagesChosen, ...packagesChosen ];
+
+        updateState(updatedPackagesAvailable, updatedPackagesChosen);
     };
 
-    // move all packages
-    const moveAll = (fromAvailable) => {
-        let chosenPackages = [];
-        if (fromAvailable) {
-            chosenPackages = [ ...packagesAvailable.filter(pack => !pack.isHidden), ...packagesChosen ];
-            setPackagesChosen(chosenPackages);
-            sortPackages([ ...packagesAvailable.filter(pack => pack.isHidden) ]);
-        } else {
-            chosenPackages = [ ...packagesChosen.filter(pack => pack.isHidden) ];
-            sortPackages([ ...packagesChosen.filter(pack => !pack.isHidden), ...packagesAvailable ]);
-            setPackagesChosen(chosenPackages);
-        }
+    const moveSelectedToAvailable = () => {
+        const newPackagesAvailable = [];
 
-        // set the steps field to the current chosen packages list
-        change(input.name, removePackagesDisplayFields(chosenPackages));
+        const updatedPackagesChosen = packagesChosen.filter((pack) => {
+            if (pack.selected) {
+                pack.selected = false;
+                pack.isHidden = false;
+                pack.name.includes(filterAvailable.current) ? newPackagesAvailable.push(pack) : null;
+                return false;
+            }
+
+            return true;
+        });
+
+        const updatedPackagesAvailable = [ ...newPackagesAvailable, ...packagesAvailable ];
+
+        updateState(updatedPackagesAvailable, updatedPackagesChosen);
+    };
+
+    const moveAllToChosen = () => {
+        const newPackagesChosen = packagesAvailable.reduce((result, pack) => {
+            pack.selected = false;
+            pack.isHidden = isHidden(filterChosen.current, pack);
+            return [ ...result, pack ];
+        }, []);
```

Comment:
Using `.map` would be better here than reduce since each step is independent of the previous ones.

Reply:
Agreed, done!